test/e2e/framework.go

```diff
@@ -1009,10 +1011,12 @@ func getImageName(uri string) string {
 // createPodOnNode creates a pod in the test namespace with a container whose type is decided by imageName.
 // Pod will be scheduled on the specified Node (if nodeName is not empty).
 // mutateFunc can be used to customize the Pod if the other parameters don't meet the requirements.
-func (data *TestData) createPodOnNode(name string, ns string, nodeName string, image string, command []string, args []string, env []corev1.EnvVar, ports []corev1.ContainerPort, hostNetwork bool, mutateFunc func(*corev1.Pod)) error {
+func (data *TestData) createPodOnNode(name string, ns string, nodeName string, image string, imageName string, command []string, args []string, env []corev1.EnvVar, ports []corev1.ContainerPort, hostNetwork bool, mutateFunc func(*corev1.Pod)) error {
```

Comment:
imageName is confusing, since this is actually use as the container name
I am not totally sure why we need this change, but if we do need it, maybe adding yet another parameter may not be the right approach. Maybe we could use `mutateFunc`?